src/plfa/part2/Confluence.lagda.md

```diff
@@ -401,105 +401,108 @@ sub-par : ∀{Γ A B} {N N′ : Γ , A ⊢ B} {M M′ : Γ ⊢ A}
 sub-par pn pm = subst-par (par-subst-zero pm) pn
 ```
 
-
 ## Parallel reduction satisfies the diamond property
 
 The heart of the confluence proof is made of stone, or rather, of
 diamond!  We show that parallel reduction satisfies the diamond
 property: that if `M ⇛ N` and `M ⇛ N′`, then `N ⇛ L` and `N′ ⇛ L` for
-some `L`.  The proof is relatively easy; it is parallel reduction's
-_raison d'etre_.
+some `L`.  The typical proof is an induction on `M ⇛ N` and `M ⇛ N′`
+so that every possible pair gives rise to a witeness `L` given by
+performing enough beta reductions in parallel.
+
+Will it be even nicer if we simply perform as many beta reductions in
+parallel as possible on `M`, say `M ⁺`, so that `N ⇛ M ⁺` holds
+independent of another `N′`? This is exactly the idea of Takahashi's
+complete development.
+ 
+```
+_⁺ : ∀ {Γ A}
+  → Γ ⊢ A → Γ ⊢ A
+(` x) ⁺       =  ` x
+(ƛ M) ⁺       = ƛ (M ⁺)
+((ƛ N) · M) ⁺ = N ⁺ [ M ⁺ ]
+(L · M) ⁺     = L ⁺ · (M ⁺)
+```
+
+The desired property may be illustrated as 
+
+    M
+    ︲\
+    ︲ \
+    ︲  \
+    ︲   N
+    ︲  /
+    ︲ /
+    ︲/
+    M⁺
+
+so that we call it the _triangle property_ which is proved by induction on `M ⇛ N` only. 
```

Comment:
I would move this diagram up, just before the definition of `_⁺`, so that it comes right after
the intuitive description of complete development.

Reply:
Done.